### ISSUE TITLE
Members not removed from the workspace when the user deleted the account

### ISSUE THREAD

**Issue by kavimuru** [Expensify/App#15267]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:

1. create workspace using userA
2. add userB as member
3. close the userB account
4. check the workspace members in userA

## Expected Result:
user should be removed from workspace members list if user delete account

## Actual Result:
user still shows in members list

## Workaround:
unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x ] Android / native
- [ x] Android / Chrome
- [x ] iOS / native
- [x ] iOS / Safari
- [x ] MacOS / Chrome / Safari
- [x ] MacOS / Desktop

**Version Number:** 1.2.74-0
**Reproducible in staging?:** y 
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** 

https://user-images.githubusercontent.com/43996225/219828515-1b1edc59-d637-4adf-a223-7edcd9d5fe58.mov

https://user-images.githubusercontent.com/43996225/219828519-76f0eac0-8af6-4b5e-96ee-8af7adee73af.mp4

**Expensify/Expensify Issue URL:**
**Issue reported by:** @gadhiyamanan 
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1676610925025799

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by MelvinBot** [comment]
Triggered auto assignment to @puneetlath (`Bug`), see https://stackoverflow.com/c/expensify/questions/14418 for more details.

**Comment by MelvinBot** [comment]
### Bug0 Triage Checklist ([Main S/O](https://stackoverflowteams.com/c/expensify/questions/14418))

- [ ] This "bug" occurs on a supported platform (ensure `Platforms` in OP are ✅)
- [ ] This bug is not a duplicate report (check [E/App issues](https://github.com/Expensify/App/issues) and [#expensify-bugs](https://expensify.slack.com/archives/C049HHMV9SM))
    - If it is, comment with a link to the original report, close the issue and add any novel details to the original issue instead
- [ ] This bug is reproducible using the reproduction steps in the OP. [S/O](https://stackoverflowteams.com/c/expensify/questions/14418#:~:text=The%20bug%20is%20reproducible)
    - If the reproduction steps are clear and you're unable to reproduce the bug, check with the reporter and QA first, then close the issue.
    - If the reproduction steps aren't clear and you determine the correct steps, please update the OP.
- [ ] This issue is filled out as thoroughly and clearly as possible
    - Pay special attention to the title, results, platforms where the bug occurs, and if the bug happens on staging/production.
- [ ] I have reviewed and subscribed to the linked Slack conversation to ensure Slack/Github stay in sync

**Comment by puneetlath** [comment]
I was able to reproduce this. The fix will need to be internal.

**Comment by MelvinBot** [comment]
Unable to auto-create job on Upwork. The BZ team member should create it manually for this issue. cc @thienlnam

**Comment by MelvinBot** [comment]
Triggered auto assignment to Contributor Plus for review of internal employee PR - @Santhosh-Sellavel (`Internal`)

**Comment by puneetlath** [comment]
Turns out there is a client-side aspect to this, so I sent a PR here: https://github.com/Expensify/App/pull/15324. 

I'm going to look into also sending a pusher update.

**Comment by MelvinBot** [comment]
Job added to Upwork: https://www.upwork.com/jobs/~01e2c97fdf9280e9e8

**Comment by MelvinBot** [comment]
Current assignee @Santhosh-Sellavel is eligible for the Internal assigner, not assigning anyone new.

**Comment by MelvinBot** [comment]
@puneetlath, @Santhosh-Sellavel Uh oh! This issue is overdue by 2 days. Don't forget to update your issues!

**Comment by puneetlath** [comment]
The PR is on staging.

**Comment by MelvinBot** [comment]
`Reviewing` label has been removed, please complete the "BugZero Checklist".

**Comment by MelvinBot** [comment]
The solution for this issue has been :rocket: deployed to production :rocket: in version 1.2.77-4 and is now subject to a 7-day regression period :calendar:. Here is the list of pull requests that resolve this issue:

- https://github.com/Expensify/App/pull/15324


If no regressions arise, payment will be issued on 2023-03-09. :confetti_ball:

After the hold period is over and BZ checklist items are completed, please complete any of the applicable payments for this issue, and check them off once done.
- [ ] External issue reporter
- [ ] Contributor that fixed the issue
- [ ] Contributor+ that helped on the issue and/or PR

As a reminder, here are the bonuses/penalties that should be applied for any External issue:
- Merged PR within 3 business days of assignment - 50% **bonus**
- Merged PR more than 9 business days after assignment - 50% **penalty**

**Comment by MelvinBot** [comment]
**BugZero Checklist:** The PR fixing this issue has been merged! The following checklist ([instructions](https://stackoverflow.com/c/expensify/questions/14574)) will need to be completed before the issue can be closed:

- [ ] [@Santhosh-Sellavel / @puneetlath] The PR that introduced the bug has been identified. Link to the PR:
- [ ] [@Santhosh-Sellavel / @puneetlath] The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment:
- [ ] [@Santhosh-Sellavel / @puneetlath] A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion:
- [ ] [@Santhosh-Sellavel] Propose [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) steps to ensure the same bug will not reach production again.
- [ ] [@puneetlath] Link the GH issue for creating/updating the regression test once above steps have been agreed upon:

**Comment by puneetlath** [comment]
@Santhosh-Sellavel I sent you a hiring offer. Can you please help with the checklist? 

@gadhiyamanan can you please apply to the upwork job for the reporting bonus: https://www.upwork.com/jobs/~01e2c97fdf9280e9e8

For me: https://www.upwork.com/ab/applicants/1628130274160820224/applicants

**Comment by Santhosh-Sellavel** [comment]
@puneetlath  I suspect this could be caught or handled [here](https://github.com/Expensify/App/pull/12746). Please leave a comment PR if you agree.

Discussion is not needed for this as it is kind of edge case, create one if you feel otherwise.

### Regression Test proposal
**Bug:**  Closed User accounts are shown on the workspace members page.
 
#### Regression steps: 

1. In device A, log in with  Account_1  (e.g admin@test.com)
2. Create a workspace if you don't already have one
3. Invite a user(Account_2) to the workspace (e.g. employee@test.com)
4. Close the right-hand modal
5. In another device B, log into Account_2 (e.g employee@test.com) and close the account
6. In device A, go back to the workspace members page
7. The Account_2 whose account was closed should no longer appear as a workspace member (it may take a second to update)

@puneetlath Believe this should be enough for the checklist, please complete the rest thanks!

**Comment by Santhosh-Sellavel** [comment]
Also accepted the offer, thanks!

**Comment by puneetlath** [comment]
Perfect, thanks @Santhosh-Sellavel. Last thing, can you start a discussion in #expensify-bugs with me, you, and the auth/reviewers of the root cause PR to see if anything could have been done differently to stop this bug from occurring in the first place?

**Comment by Santhosh-Sellavel** [comment]
@puneetlath Discussion Link https://expensify.slack.com/archives/C049HHMV9SM/p1678308308662289

**Comment by puneetlath** [comment]
Paid. Thanks everyone!
